source/exe/BUILD

```diff
@@ -100,7 +95,10 @@ envoy_cc_library(
         "//source/server:hot_restart_nop_lib",
         "//source/server:proto_descriptors_lib",
         "//source/server/config_validation:server_lib",
-    ],
+    ] + select({
+        "//bazel:disable_signal_trace": [],
```

Comment:
Great idea.  Doing so now.  Actually, why is this a compile-time decision rather than an option?  Then it would be easier to test without rebuilding the entire tree.  Same with HotRestart, though that's addressed in https://github.com/envoyproxy/envoy/pull/2568.

Reply:
Some of these options are compile-time as we want to squash dependencies (e.g. Google gRPC) or demonstrate that site-local replacement or build time removal of features is possible (e.g. we replace during import on the Google side). Hot restart is probably a better contender to be fully dynamic, as it doesn't introduce any deps.

There's a wider conversation happening right now in https://github.com/envoyproxy/envoy/issues/2069. Basically, we need some way to provide greater build time flexibility to allow very minimal Envoy binary deployments while also support kitchen sink builds with dynamic configuration. The former might be interesting in resource constrained embedded or container environments, the latter on a traditional fat server.

Reply:
I wasn't arguing against *any* compile-time switches, but it wasn't obvious what dependency this one had.  I was thinking that if we were compiling on Windows and the APIs didn't exist, we'd just #if them out based on platform support as opposed to config, but keep the class API above that consistent.

Reply:
Yeah, the `signal_trace` deps are the Backtrace library, an external dependency.